### PR TITLE
chore(useDeviceOrientation): adds event target

### PR DESCRIPTION
**Pull request by anteriovieira**
@antfu what do you think about it?

image

the demo currently does not display the values because it does not have context access

### CODE REVIEW

File: packages/core/useDeviceOrientation/index.ts [hunk 1]
```diff
@@ -1,20 +1,24 @@
 /* this implementation is original ported from https://github.com/logaretm/vue-use-web by Abdelrahman Awad */
 
+import { isClient } from '@vueuse/shared'
 import { ref, Ref } from 'vue-demi'
 import { useEventListener } from '../useEventListener'
 
-export function useDeviceOrientation() {
+export function useDeviceOrientation(target: Window = window) {
```
Comment (antfu) [inline packages/core/useDeviceOrientation/index.ts:7]:
```suggestion
export function useDeviceOrientation(target: Window = isClient ? window : null) {
```

SSR will only work this way

File: packages/core/useDeviceOrientation/index.ts [hunk 2]
```diff
@@ -1,20 +1,24 @@
 /* this implementation is original ported from https://github.com/logaretm/vue-use-web by Abdelrahman Awad */
 
+import { isClient } from '@vueuse/shared'
 import { ref, Ref } from 'vue-demi'
 import { useEventListener } from '../useEventListener'
 
-export function useDeviceOrientation() {
+export function useDeviceOrientation(target: Window | null = isClient ? window : null) {
+  const eventTarget = isClient ? target : null
```
Comment (antfu) [inline packages/core/useDeviceOrientation/index.ts:8]: Let's just use target

Comment (anteriovieira) [reply packages/core/useDeviceOrientation/index.ts:8]: I was analyzing here, we have many places that would need this implementation. Maybe not the best way. And I will think about it better.

Comment (anteriovieira) [reply packages/core/useDeviceOrientation/index.ts:8]: For example: `useDeviceMotion`, `useDeviceLIght`

Comment (antfu) [reply packages/core/useDeviceOrientation/index.ts:8]: Yeah, it's better to have an option for it. Will also be more extensible for us to add more options in the future.

I think we can make a shared interface with shared js-docs (providing more details/explication for users) like

```ts
export interface ConfigurableWindow {
  /*
   * Specify a custom `window` instance, e.g. working with iframes or in testing envroiments.
   * ...(more)
   */
  window?: Window
}

export interface ConfigurableDocument {
  /*
   * Specify a custom `document` instance, e.g. working with iframes or in testing envroiments.
   * ...(more)
   */
  document?: Document
}
```

```ts
interface DeviceMotionOptions extends ConfigurableWindow {
  throttleMs: number
}

export function useDeviceMotion(options: DeviceMotionOptions = { throttleMs: 10 }) {
  // options.window
  /* ... */
}
```

Comment (antfu) [reply packages/core/useDeviceOrientation/index.ts:8]: `navigator`, `location` as well. Any global reference in functions should be configurable then I think.

Comment (anteriovieira) [reply packages/core/useDeviceOrientation/index.ts:8]: I agree with you, I was looking for a solution as a global configuration, but I think your idea will give more flexibility for the use of the functions.

Comment (antfu) [reply packages/core/useDeviceOrientation/index.ts:8]: I will do the refactor 👍

**Pull request closed without merging**